lib/models/environment-variables.coffee

```diff
@@ -158,22 +159,23 @@ exports.device = {}
 # @public
 # @function
 #
-# @param {(String|Number)} deviceId - device id
+# @param {String} deviceName - device name
 # @param {module:resin.models.environment-variables.device~getAllCallback} callback - callback
 #
 # @example
-#	resin.models.environmentVariables.device.getAll 51, (error, environmentVariables) ->
+#	resin.models.environmentVariables.device.getAll 'MyDevice', (error, environmentVariables) ->
 #		throw error if error?
 #		console.log(environmentVariables)
 ###
-exports.device.getAll = (deviceId, callback) ->
-	return pine.get
-		resource: 'device_environment_variable'
-		options:
-			filter:
-				device: deviceId
-			expand: 'device'
-			orderby: 'env_var_name asc'
+exports.device.getAll = (deviceName, callback) ->
+	deviceModel.get(deviceName).then (device) ->
```

Comment:
@Page- `.get` still needs to be able to get the whole device as it's a public function. The only thing I can think of is implementing a `.getIDByDeviceName()` function or something like that, and reuse that one in these two functions. What do you think?